house_robber.java

```diff
@@ -0,0 +1,30 @@
+// Time Complexity : O(n), where n is the length of the nums array
+// Space Complexity :O(1), as we are just using 3 constant variables
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this : did not understand to do this problen with single dp array (and "not" dp[nums.length][2])
```

Comment:
Using single dimensional dp array is exactly the same as this except for the fact that only the 2 most recent indices (i - 1, i - 2) are only used